resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/security/HadoopFSCredentialProvider.scala

```diff
@@ -48,9 +50,16 @@ private[security] class HadoopFSCredentialProvider
     val tmpCreds = new Credentials()
     val tokenRenewer = getTokenRenewer(hadoopConf)
     hadoopFSsToAccess(hadoopConf, sparkConf).foreach { dst =>
-      val dstFs = dst.getFileSystem(hadoopConf)
-      logInfo("getting token for: " + dst)
-      dstFs.addDelegationTokens(tokenRenewer, tmpCreds)
+      try {
+        val dstFs = dst.getFileSystem(hadoopConf)
+        logInfo("getting token for: " + dst)
+        dstFs.addDelegationTokens(tokenRenewer, tmpCreds)
+      } catch {
+        case e: StandbyException =>
+          logWarning(s"Namenode ${dst} is in state standby", e)
```

Comment:
In our tests, there are two possible exceptions when yarn.spark.access.namenodes=hdfs://activeNamenode,hdfs://standbyNamenode
1) Caused by: org.apache.hadoop.ipc.RemoteException(org.apache.hadoop.ipc.StandbyException): Operation category WRITE is not supported in state standby
2) Caused by: org.apache.hadoop.ipc.StandbyException: Operation category WRITE is not supported in state standby
Maybe RemoteException should be caught by better way.